Shopify/cli
# App Watcher: support gitignore files and watch_paths in toml

### WHY are these changes introduced?
Until now, in the new AppWatcher, we were watching for any change in the extensions folders. This works fine for most cases, but can be tricky for extensions that autogenerate or autoupdate files during the build process, as it could enter in a build-watch loop.



### WHAT is this pull request doing?
-  By default, we watch for any change in the extensions folder except for the command ignored patterns (like `node_modules`
- If an extension folder has a `.gitignore`, change in the files included in gitignore will be ignored.
- If an extension toml contains a configuration to watch specific paths, then only those paths will be watched. (this is possible for functions for now)



### How to test your changes?
Having an app with a function in Rust.
 Try running `shopify app dev` (with the management API) and make changes in the extension, for any change, only one update should be triggered. 
Check that changes in files defined in the gitignore are not triggering an update. 



### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
